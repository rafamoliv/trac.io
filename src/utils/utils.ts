/**
 * Util to formate date to locale string
 *
 * @param {Date} DateToFormat
 * @returns '14/12/2022, 21:00:00'
 */

export const dateFormat = (DateToFormat: Date) => {
  if (!DateToFormat) return
  const date = new Date(DateToFormat)
  return date.toLocaleString()
}

/**
 * Util to return initial letters from name and lastname
 *
 * @param {string} name
 * @returns 'RO'
 */

export const initialLettersNameLastname = (name: string) => {
  if (!name) return
  return name
    .match(/(\b\S)?/g)
    .join('')
    .toUpperCase()
}
