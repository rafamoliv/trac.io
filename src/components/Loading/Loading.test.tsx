import React from 'react'
import i18next from 'i18next'
import { expect, it } from 'vitest'

import { render } from '../../utils/tests'

import { Loading } from '.'

const props = {
    children: i18next.t('translation:foo'),
    loading: true
}

describe('<Loading />', () => {
    describe('Component Render', () => {
        it('render element', () => {
            const { container } = render(<Loading {...props} />)
            expect(container.firstChild).toMatchSnapshot()
        })
    })
})
