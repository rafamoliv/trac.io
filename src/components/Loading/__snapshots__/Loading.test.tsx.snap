// Vitest Snapshot v1

exports[`<Loading /> > Component Render > render element 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
}

<div
  class="c0"
>
  <div
    aria-label="animation"
    role="button"
    style="width: 68px; height: 68px; overflow: hidden; margin: 0px auto; outline: none;"
    tabindex="0"
    title=""
  >
    <svg
      height="100"
      preserveAspectRatio="xMidYMid meet"
      style="width: 100%; height: 100%; transform: translate3d(0,0,0);"
      viewBox="0 0 100 100"
      width="100"
      xmlns="http://www.w3.org/2000/svg"
      xmlns:xlink="http://www.w3.org/1999/xlink"
    >
      <defs>
        <clippath
          id="__lottie_element_2"
        >
          <rect
            height="100"
            width="100"
            x="0"
            y="0"
          />
        </clippath>
        <g
          id="__lottie_element_10"
        >
          <g
            opacity="1"
            style="display: block;"
            transform="matrix(0.5,0,0,0.5,-75,82.75)"
          >
            <rect
              fill="#000000"
              height="500"
              width="500"
            />
          </g>
        </g>
        <filter
          filterUnits="objectBoundingBox"
          height="100%"
          id="__lottie_element_16"
          width="100%"
          x="0%"
          y="0%"
        >
          <fecomponenttransfer
            in="SourceGraphic"
          >
            <fefunca
              tableValues="1.0 0.0"
              type="table"
            />
          </fecomponenttransfer>
        </filter>
        <mask
          id="__lottie_element_10_2"
          mask-type="alpha"
        >
          <g
            filter="url(#__lottie_element_16)"
          >
            <rect
              fill="#ffffff"
              height="100"
              opacity="0"
              width="100"
              x="0"
              y="0"
            />
            <use
              href="#__lottie_element_10"
            />
          </g>
        </mask>
      </defs>
      <g
        clip-path="url(#__lottie_element_2)"
      >
        <g
          mask="url(#__lottie_element_10_2)"
          style="display: block;"
        >
          <g
            opacity="1"
            transform="matrix(0.07088448107242584,0.07053644955158234,-0.07053644955158234,0.07088448107242584,50,6.5)"
          >
            <g
              opacity="1"
              transform="matrix(1,0,0,1,0,-0.25)"
            >
              <path
                d=" M0,-55 C30.35449981689453,-55 55,-30.35449981689453 55,0 C55,30.35449981689453 30.35449981689453,55 0,55 C-30.35449981689453,55 -55,30.35449981689453 -55,0 C-55,-30.35449981689453 -30.35449981689453,-55 0,-55z"
                fill="rgb(116,165,232)"
                fill-opacity="1"
              />
              <path
                d=" M0,-55 C30.35449981689453,-55 55,-30.35449981689453 55,0 C55,30.35449981689453 30.35449981689453,55 0,55 C-30.35449981689453,55 -55,30.35449981689453 -55,0 C-55,-30.35449981689453 -30.35449981689453,-55 0,-55z"
                fill-opacity="0"
                stroke="rgb(255,255,255)"
                stroke-linecap="butt"
                stroke-linejoin="miter"
                stroke-miterlimit="4"
                stroke-opacity="1"
                stroke-width="0"
              />
            </g>
          </g>
        </g>
        <g
          style="display: none;"
        >
          <g>
            <path />
            <path
              fill-opacity="0"
              stroke-linecap="butt"
              stroke-linejoin="miter"
              stroke-miterlimit="4"
            />
          </g>
        </g>
        <g
          opacity="1"
          style="display: block;"
          transform="matrix(0.41467416286468506,0.41263821721076965,-0.41263821721076965,0.41467416286468506,50,51)"
        >
          <g
            opacity="1"
            transform="matrix(1,0,0,1,0,-0.25)"
          >
            <path
              d=" M0,-55 C30.35449981689453,-55 55,-30.35449981689453 55,0 C55,30.35449981689453 30.35449981689453,55 0,55 C-30.35449981689453,55 -55,30.35449981689453 -55,0 C-55,0 -55,0 -55,0"
              fill-opacity="0"
              stroke="rgb(0,0,0)"
              stroke-linecap="butt"
              stroke-linejoin="miter"
              stroke-miterlimit="4"
              stroke-opacity="1"
              stroke-width="3"
            />
          </g>
        </g>
      </g>
    </svg>
  </div>
</div>
`;
